core/src/price.ts

```diff
@@ -19,7 +19,11 @@ const calculateElapsedSteps = function (auction: Auction, currentDate: Date): nu
     checkAuctionStartDate(auction.startDate, currentDate);
     const auctionStartTimestamp = auction.startDate.getTime();
     const elapsedTime = (currentDate.getTime() - auctionStartTimestamp) / 1000;
-    return Math.floor(elapsedTime / auction.secondsBetweenPriceDrops);
+    const distance = elapsedTime / auction.secondsBetweenPriceDrops;
+    const steps = Math.floor(distance);
+    // already predict for the next step when the next step is about to reached
+    const compensation = distance - steps > 0.99 ? 1 : 0;
+    return steps + compensation;
```

Comment:
Am I reading this correctly: this change drops the price of the auction 1% of the `now - auctionStart` time? So the longer the auction lives the earlier one step compensation will apply? Technically, in the end, it can be bigger than the step itself, eg 2 hours after the start the step will be applied 72 seconds earlier? I don't think it's correct, sounds super random adjustment

Moreover, I think that dropping price earlier shouldn't be correct. If this fix only helps reduce the margin, then we should not apply it at all. If this fix helps making more auctions in the "executable state" on our side, you can instead warp time before execution to make sure that all auctions are profitable. Better be more careful (later apply cut) than earlier